includes/woopay/class-woopay-session.php

```diff
@@ -337,15 +337,66 @@ public static function get_frontend_init_session_request() {
 		return WooPay_Utilities::encrypt_and_sign_data( $session );
 	}
 
+	/**
+	 * Retrieves cart data from the current session.
+	 *
+	 * If the request doesn't come from WooPay, this uses the same strategy in
+	 * `hydrate_from_api` on the Checkout Block to retrieve cart data.
+	 *
+	 * @param int|null $order_id Pay-for-order order ID.
+	 * @param string|null $key Pay-for-order key.
+	 * @param string|null $billing_email Pay-for-order billing email.
+	 * @param WP_REST_Request|null $woopay_request The WooPay request object.
+	 * @return array The cart data.
+	 */
+	private static function get_cart_data( $is_pay_for_order, $order_id, $key, $billing_email, $woopay_request ) {
```

Comment:
I extracted the implementations of the cart and checkout requests from the main function as this logic is getting bigger. This way these and the main functions gets easier to consume and more legible.